controllers/suite_test.go

```diff
@@ -37,6 +43,14 @@ import (
 var cfg *rest.Config
 var k8sClient client.Client
 var testEnv *envtest.Environment
+var mgr manager.Manager
```

Comment:
Wouldn't it be better to avoid these global variables and use local variables in the test functions instead?

This applies to the other global variables too.

Reply:
The manager creates and starts the controller. I thought to use the same instance for all the tests. As these tests run in parallel, I ran into glitches when I ran the manager locally in the tests.

Reply:
> As these tests run in parallel

That was my concern. If they run in parallel, won't the test preparation data and results conflict?

Reply:
I create different etcd resources for each test and clean them up once completed. so they wont interfere with each other.